src/D2L.CodeStyle.Analyzers/ApiUsage/JsonParamBinderAttribute/JsonParamBinderAnalyzerFixer.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+namespace D2L.CodeStyle.Analyzers.ApiUsage.JsonParamBinderAttribute {
+
+	[ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(JsonParamBinderAnalyzerFixer) )]
+	public class JsonParamBinderAnalyzerFixer : CodeFixProvider {
+
+		public override FixAllProvider GetFixAllProvider() {
+			return WellKnownFixAllProviders.BatchFixer;
+		}
+
+		public override async Task RegisterCodeFixesAsync( CodeFixContext context ) {
+			Diagnostic diagnostic = context.Diagnostics.First();
+			TextSpan sourceSpan = diagnostic.Location.SourceSpan;
+
+			SyntaxNode root = await context.Document.GetSyntaxRootAsync(context.CancellationToken);
+
+			AttributeSyntax oldAttribute = root.FindToken(sourceSpan.Start)
+				.Parent.AncestorsAndSelf().OfType<AttributeSyntax>().First();
+
+			context.RegisterCodeFix( 
+				CodeAction.Create( 
+					Diagnostics.ObsoleteJsonParamBinder.Title.ToString(),
+					ct => SwapAttributes( context.Document, oldAttribute, ct)
+				), 
+				diagnostic
+			);
+		}
+
+		private async Task<Document> SwapAttributes( Document document, AttributeSyntax oldAttribute, CancellationToken ct ) {

```

Comment:
Don't need to change it in this PR (I'll eventually hit it up in some future formatting pass) but we typically stick to 80 columns in this repo. GitHub's diffs are super annoying to read if there is horizontal scrolling.

I'll look into the built-in VS2017 `.editorconfig` stuff for this repo today.

Reply:
I may as well fix it

Reply:
That's cool, but the D2L standard/this repo is 80 :P
(120 causes problems for me with split view in VS, but that may be because of my font-size.)